### DISCUSSION TITLE
How can i auto calculator the demension of jpeg from jpeg get wrong demension

### DISCUSSION THREAD

**Discussion by HanalogInstruments** [ImageMagick/ImageMagick#5237] [Help]
Hi,

I have jpeg encrypted all header but only raw data and i patch the same chroma sampling and jpeg working but came with header demension and how can i calculate the demension from raw data to make is working again?

Image Preview:
![Preview](https://user-images.githubusercontent.com/100456607/174122663-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg)

Original jpg:
https://www.mediafire.com/file/q2oe8vl3sl3l7nm/Sample.jpg/file

**Comment by snibgo** [comment]
MediaFire says I can't download that image unless I permit MediaFire to steal my data, which I won't.

**Reply by HanalogInstruments** [reply]
> MediaFire says I can't download that image unless I permit MediaFire to steal my data, which I won't.
> 
> 

Is clean file if you don't want to download the original file you can use the image i upload for preview to check

**Comment by snibgo** [comment]
IM says that preview file is corrupt.
```
magick 174122663-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg info:

magick: Corrupt JPEG data: 2915448 extraneous bytes before marker 0xd9 `17412266
3-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg' @ warning/jpeg.c/JPEGWarningHandler/403.
174122663-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg JPEG 1024x768 1024x768+0+0 8-bit sRGB 2.96501MiB 0.000u 0:00.002
```

**Reply by HanalogInstruments** [reply]
> IM says that preview file is corrupt.
> 
> ```
> 
> magick 174122663-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg info:
> 
> 
> 
> magick: Corrupt JPEG data: 2915448 extraneous bytes before marker 0xd9 `17412266
> 
> 3-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg' @ warning/jpeg.c/JPEGWarningHandler/403.
> 
> 174122663-bf7c1419-5774-40b8-a923-1c6dd0bc2064.jpg JPEG 1024x768 1024x768+0+0 8-bit sRGB 2.96501MiB 0.000u 0:00.002
> 
> ```
> 
> 

Yes, the preview is corrupted because the demension is from
Another header file and i glue the raw data on it with same chroma sampling and standard tables 100%, so i want to know how calculate the demesion to fix this image via imagegick

**Comment by snibgo** [comment]
IM doesn't have facilities to fix corrupt JPEG files.
